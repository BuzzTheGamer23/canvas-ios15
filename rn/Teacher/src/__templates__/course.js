//
// Copyright (C) 2016-present Instructure, Inc.
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, version 3 of the License.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with this program.  If not, see <http://www.gnu.org/licenses/>.
//

/* @flow */

import template, { type Template } from '../utils/template'
import { section } from './section'

export const course: Template<Course> = template({
  id: '1',
  name: 'Learn React Native',
  short_name: 'rn',
  course_code: 'rn 101',
  image_download_url: 'https://farm3.staticflickr.com/2926/14690771011_945f91045a.jpg',
  is_favorite: true,
  default_view: 'wiki',
  term: { name: 'Default Term' },
  enrollments: [{
    enrollment_state: 'active',
    role: 'TeacherEnrollment',
    role_id: '1',
    type: 'teacher',
    user_id: '1',
  }],
})

export const courseWithSection: Template<Course> = function (defaults) {
  return course({
    ...defaults,
    sections: [section()],
  })
}

export const customColors: Template<CustomColors> = template({
  custom_colors: {
    course_1: '#fff',
  },
})
