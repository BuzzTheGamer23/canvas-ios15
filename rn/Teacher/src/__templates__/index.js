// @flow

export * from '../redux/__templates__/app-state'
export * from './account-notification'
export * from './addressBook'
export * from './api'
export * from './assignments'
export * from './attachment'
export * from './conversations'
export * from './course'
export * from './discussion'
export * from './enrollments'
export * from './error'
export * from './external-tool'
export * from './file'
export * from './folder'
export * from './grading-periods'
export * from './group'
export * from './helm'
export * from './launch-definitions'
export * from './mediaComment'
export * from './page'
export * from './quiz'
export * from './quizSubmission'
export * from './role'
export * from './rubric'
export * from './section'
export * from './session'
export * from './submissions'
export * from './tab'
export * from './toDo'
export * from './users'
