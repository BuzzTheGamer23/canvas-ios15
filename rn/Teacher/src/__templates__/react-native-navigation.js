// @flow

import template, { type Template } from '../utils/template'

export const navigator: Template<ReactNavigator> = template({
  push: jest.fn(),
  showModal: jest.fn(),
  dismissModal: jest.fn(),
  setOnNavigatorEvent: jest.fn(),
  pop: jest.fn(),
  setTitle: jest.fn(),
  setStyle: jest.fn(),
})
