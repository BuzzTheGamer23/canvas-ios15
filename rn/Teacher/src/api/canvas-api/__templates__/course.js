/* @flow */

import template, { type Template } from '../../../utils/template'

export const course: Template<Course> = template({
  id: '1',
  name: 'Learn React Native',
  short_name: 'rn',
  course_code: 'rn 101',
  image_download_url: 'https://farm3.staticflickr.com/2926/14690771011_945f91045a.jpg',
  is_favorite: true,
  default_view: 'wiki',
  term: { name: 'Default Term' },
})

export const customColors: Template<CustomColors> = template({
  custom_colors: {
    course_1: '#fff',
  },
})
