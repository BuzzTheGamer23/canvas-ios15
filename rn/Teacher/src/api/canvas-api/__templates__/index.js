/* @flow */

export default {
  ...require('./course'),
}
