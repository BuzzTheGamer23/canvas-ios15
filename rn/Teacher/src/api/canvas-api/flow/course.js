// @flow

export type Course = {
  id: string,
  name: string,
  course_code: string,
  short_name?: string,
  image_download_url?: ?string,
  is_favorite?: boolean,
}

export type CustomColors = {
  custom_colors: {
    [string]: string,
  },
}

export type UpdateCustomColorResponse = {
  hexcode: string,
}

export type Favorite = {
  context_id: string,
  context_type: string,
}
