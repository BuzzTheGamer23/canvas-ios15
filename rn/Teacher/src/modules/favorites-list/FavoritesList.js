// @flow

import React, { Component } from 'react'
import ReactNative, {
  ListView,
  StyleSheet,
} from 'react-native'
import { connect } from 'react-redux'
import i18n from 'format-message'

import CourseFavorite from './components/CourseFavorite'
import FavoritesActions from './actions'
import mapStateToProps from './map-state-to-props'

type Props = {
  navigator: ReactNavigator,
  courses: Array<Course>,
  toggleFavorite: (courseId: string, favorite: boolean) => Promise<*>,
}

type State = {
  ds: ReactNative.ListViewDataSource,
  dataSource: ReactNative.ListViewDataSource,
}

export class FavoritesList extends Component {
  props: Props
  state: State

  static navigatorButtons = {
    rightButtons: [{
      title: i18n({
        default: 'Done',
        description: 'Back button to move back a screen',
      }),
      id: 'done',
      testId: 'done_button',
    }],
  }

  constructor (props: Props) {
    super(props)

    let ds = new ListView.DataSource({
      rowHasChanged: (r1, r2) => r1 !== r2,
    })

    this.state = {
      ds,
      dataSource: ds.cloneWithRows(this.props.courses),
    }

    props.navigator.setOnNavigatorEvent(this.onNavigatorEvent)
  }

  componentWillReceiveProps (nextProps: Props) {
    this.setState({
      dataSource: this.state.ds.cloneWithRows(nextProps.courses),
    })
  }

  onNavigatorEvent = (event: NavigatorEvent) => {
    if (event.type === 'NavBarButtonPress') {
      if (event.id === 'done') {
        this.props.navigator.dismissModal({
          animationType: 'slide-down',
        })
      }
    }
  }

  renderCourse = (course: Course) => {
    return (
      <CourseFavorite
        course={course}
        onPress={this.props.toggleFavorite}
      />
    )
  }

  render (): React.Element<ListView> {
    return (
      <ListView
        style={styles.listStyle}
        dataSource={this.state.dataSource}
        renderRow={this.renderCourse}
      />
    )
  }
}

let connected = connect(mapStateToProps, FavoritesActions)(FavoritesList)
export default (connected: FavoritesList)

const styles = StyleSheet.create({
  listStyle: {
    backgroundColor: '#fff',
  },
})
