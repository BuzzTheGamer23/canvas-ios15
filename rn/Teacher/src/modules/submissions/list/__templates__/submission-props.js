// @flow

import template, { type Template } from '../../../../utils/template'
import type { SubmissionDataProps } from '../submission-prop-types'

const templates = {
  ...require('../../../../__templates__/submissions'),
}

const submissionWithHistory = templates.submissionHistory()

export const submissionProps: Template<SubmissionDataProps> = template({
  grade: 'B-',
  name: submissionWithHistory.user.name,
  status: 'submitted',
  userID: submissionWithHistory.user.id,
  avatarURL: submissionWithHistory.user.avatar_url,
})
