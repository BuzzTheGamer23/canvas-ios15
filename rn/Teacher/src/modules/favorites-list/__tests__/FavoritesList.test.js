// @flow

import 'react-native'
import React from 'react'
import * as courseTemplate from '../../../api/canvas-api/__templates__/course'
import * as navigationTemplate from '../../../__templates__/react-native-navigation'
import { FavoritesList } from '../FavoritesList'

import renderer from 'react-test-renderer'

let courses = [
  courseTemplate.course({ is_favorite: false }),
  courseTemplate.course({ is_favorite: true }),
]

let defaultProps = {
  navigator: navigationTemplate.navigator(),
  courses: courses,
  toggleFavorite: () => Promise.resolve(),
}

test('renders correctly', () => {
  let tree = renderer.create(
    <FavoritesList {...defaultProps} />
  ).toJSON()

  expect(tree).toMatchSnapshot()
})

test('calls dismissModal when back button is selected', () => {
  let navigator = navigationTemplate.navigator({
    dismissModal: jest.fn(),
  })

  let tree = renderer.create(
    <FavoritesList {...defaultProps} navigator={navigator} />
  )

  tree._component._renderedComponent._instance.onNavigatorEvent({
    type: 'NavBarButtonPress',
    id: 'done',
  })

  expect(navigator.dismissModal).toHaveBeenCalledWith({
    animationType: 'slide-down',
  })
})
