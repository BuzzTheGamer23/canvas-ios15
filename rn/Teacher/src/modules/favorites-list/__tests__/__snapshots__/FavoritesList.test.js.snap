// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View>
  <RCTScrollView>
    <View>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
            },
            Object {
              "borderBottomColor": "#e3e3e3",
              "borderBottomWidth": 1,
            },
          ]
        }
        testID="rn 101"
        tvParallaxProperties={undefined}
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "white",
              "flexDirection": "row",
              "paddingHorizontal": 16,
              "paddingVertical": 16,
            }
          }
        >
          <Image
            source={1}
            style={
              Array [
                Object {
                  "height": 20,
                  "marginRight": 10,
                  "width": 20,
                },
                Object {
                  "tintColor": "#374A59",
                },
              ]
            }
          />
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "fontSize": 16,
                "fontWeight": "400",
                "marginRight": 16,
              }
            }
          >
            Learn React Native
          </Text>
        </View>
      </View>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
            },
            Object {
              "borderBottomColor": "#e3e3e3",
              "borderBottomWidth": 1,
            },
          ]
        }
        testID="rn 101"
        tvParallaxProperties={undefined}
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "white",
              "flexDirection": "row",
              "paddingHorizontal": 16,
              "paddingVertical": 16,
            }
          }
        >
          <Image
            source={1}
            style={
              Array [
                Object {
                  "height": 20,
                  "marginRight": 10,
                  "width": 20,
                },
                Object {
                  "tintColor": "#374A59",
                },
              ]
            }
          />
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "fontSize": 16,
                "fontWeight": "400",
                "marginRight": 16,
              }
            }
          >
            Learn React Native
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
